src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { css } from 'styled-components';
+
+import { Box, Icon, Text } from '@/components';
+import { HeadingBlock, useEditorStore } from '@/features/docs/doc-editor';
+import { MAIN_LAYOUT_ID } from '@/layouts/conf';
+
+import { Heading } from './Heading';
+
+type DocSummaryProps = {
+  headings: HeadingBlock[];
+};
+
+export const DocSummary = ({ headings }: DocSummaryProps) => {
```

Comment:
I initially called the feature `doc-summary`, then it was advised to change it to "Table of Content.", let's keep it like that so.
https://github.com/numerique-gouv/impress/commit/5bd78b80687a92818497df7d443f7bac252a1067


```suggestion
export const TableContent = ({ headings }: TableContentProps) => {
```